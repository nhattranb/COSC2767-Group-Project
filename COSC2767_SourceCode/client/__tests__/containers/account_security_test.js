import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import AccountSecurity from '../../app/containers/AccountSecurity';

const mockStore = configureStore([]);

describe('AccountSecurity component', () => {
  let store;
  
  const initialState = {
    account: {
      user: {
        firstName: 'John',
        lastName: 'Doe'
      }
    },
    resetPassword: {
      resetFormData: {
        password: '',
        confirmPassword: ''
      },
      formErrors: {}
    }
  };

  beforeEach(() => {
    store = mockStore(initialState);
  });

  it('renders correctly with empty form', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <AccountSecurity />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with form errors', () => {
    const stateWithErrors = {
      ...initialState,
      resetPassword: {
        ...initialState.resetPassword,
        formErrors: {
          password: 'Password is required',
          confirmPassword: 'Passwords do not match'
        }
      }
    };
    store = mockStore(stateWithErrors);

    const { asFragment } = render(
      <Provider store={store}>
        <AccountSecurity />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
