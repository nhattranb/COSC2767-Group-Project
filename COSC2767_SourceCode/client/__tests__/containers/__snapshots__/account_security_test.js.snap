// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountSecurity component renders correctly with empty form 1`] = `
<DocumentFragment>
  <div
    class="account-security"
  >
    <div
      class="sub-page"
    >
      <div
        class="subpage-header"
      >
        <h3
          class="mb-0"
        >
          Account Security
        </h3>
      </div>
      <div
        class="subpage-body"
      >
        <div
          class="reset-form"
        >
          <h4>
            Reset Password
          </h4>
          <div
            class="reset-password-form"
          >
            <form
              novalidate=""
            >
              <div
                class="row"
              >
                <div
                  class="col-12 col-lg-6"
                >
                  <div
                    class="input-box "
                  >
                    <label>
                      Password
                    </label>
                    <div
                      class="input-text-block"
                    >
                      <input
                        autocomplete="on"
                        class="input-text"
                        name="password"
                        placeholder="Old Password"
                        type="password"
                        value=""
                      />
                    </div>
                    <span
                      class="invalid-message"
                    />
                  </div>
                </div>
                <div
                  class="col-12 col-lg-6"
                >
                  <div
                    class="input-box "
                  >
                    <label>
                      Confirm Password
                    </label>
                    <div
                      class="input-text-block"
                    >
                      <input
                        autocomplete="on"
                        class="input-text"
                        name="confirmPassword"
                        placeholder="Confirm Password"
                        type="password"
                        value=""
                      />
                    </div>
                    <span
                      class="invalid-message"
                    />
                  </div>
                </div>
              </div>
              <hr />
              <div
                class="reset-actions"
              >
                <button
                  class="input-btn custom-btn-secondary md text-only icon-left "
                  style="border-radius: 3px;"
                  type="submit"
                >
                  <span
                    class="btn-text"
                  >
                    Reset Password
                  </span>
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`AccountSecurity component renders correctly with form errors 1`] = `
<DocumentFragment>
  <div
    class="account-security"
  >
    <div
      class="sub-page"
    >
      <div
        class="subpage-header"
      >
        <h3
          class="mb-0"
        >
          Account Security
        </h3>
      </div>
      <div
        class="subpage-body"
      >
        <div
          class="reset-form"
        >
          <h4>
            Reset Password
          </h4>
          <div
            class="reset-password-form"
          >
            <form
              novalidate=""
            >
              <div
                class="row"
              >
                <div
                  class="col-12 col-lg-6"
                >
                  <div
                    class="input-box invalid"
                  >
                    <label>
                      Password
                    </label>
                    <div
                      class="input-text-block"
                    >
                      <input
                        autocomplete="on"
                        class="input-text"
                        name="password"
                        placeholder="Old Password"
                        type="password"
                        value=""
                      />
                    </div>
                    <span
                      class="invalid-message"
                    >
                      P
                    </span>
                  </div>
                </div>
                <div
                  class="col-12 col-lg-6"
                >
                  <div
                    class="input-box invalid"
                  >
                    <label>
                      Confirm Password
                    </label>
                    <div
                      class="input-text-block"
                    >
                      <input
                        autocomplete="on"
                        class="input-text"
                        name="confirmPassword"
                        placeholder="Confirm Password"
                        type="password"
                        value=""
                      />
                    </div>
                    <span
                      class="invalid-message"
                    >
                      P
                    </span>
                  </div>
                </div>
              </div>
              <hr />
              <div
                class="reset-actions"
              >
                <button
                  class="input-btn custom-btn-secondary md text-only icon-left "
                  style="border-radius: 3px;"
                  type="submit"
                >
                  <span
                    class="btn-text"
                  >
                    Reset Password
                  </span>
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
